src/black/strings.py

```diff
@@ -142,6 +152,11 @@ def normalize_string_prefix(s: str, remove_u_prefix: bool = False) -> str:
     return f"{new_prefix}{match.group(2)}"
 
 
+@lru_cache(maxsize=256)
```

Comment:
Does this really help? `re.compile` already caches things internally I believe.

Reply:
Looking at the source and testing locally, I can confirm that both re and regex cache compiled patterns. Somehow I still observe a 8-9% speed up on that strings list literal file. Running this test script:
```python
import time
from functools import lru_cache

import re
import regex

_cached_regex_compile = lru_cache(regex.compile)

# This is *roughly* how many times regex.compile was called formatting strings-list.py
loops = 410 
pattern = r"\s*\t+\s*(\S)"

re.compile(pattern)
regex.compile(pattern)

t0 = time.perf_counter()
for i in range(1, loops + 1):
    re.compile(pattern)
elapsed = time.perf_counter() - t0
print(f"{loops} cached compiles with re: {elapsed * 1000:.1f}ms")

t0 = time.perf_counter()
for i in range(1, loops + 1):
    regex.compile(pattern)
elapsed = time.perf_counter() - t0
print(f"{loops} cached compiles with regex: {elapsed * 1000:.1f}ms")

t0 = time.perf_counter()
for i in range(1, loops + 1):
    _cached_regex_compile(pattern)
elapsed = time.perf_counter() - t0
print(f"{loops} cached compiles with regex + lru_cache: {elapsed * 1000:.1f}ms")
```
I get:
- 410 cached compiles with re: 0.4ms
- 410 cached compiles with regex: 1.6ms
- 410 cached compiles with regex + lru_cache: 0.1ms

So it seems like this is the source of the slight speedup, but oddly enough this only accounts for ~half of the 8-9% speedup I saw for fmt-strings-list. Maybe the fact this script only repeated compiles one regex is why the delta isn't 1:1. For real life cases this speedup is probably minimal so I wouldn't mind reverting this to be less confusing :)

Reply:
Interesting, thanks for the extra information. I looked at `re.compile` and it calls a helper function (`_compile`) and does an `isinstance` call before looking at the cache, and the cache key is a tuple `type(pattern), pattern, flags`. I suppose that adds up to more overhead than whatever lru_cache does internally.